agent/acs/handler/acs_handler.go

```diff
@@ -492,6 +505,21 @@ func newDisconnectionTimer(client wsclient.ClientServer, timeout time.Duration,
 	return timer
 }
 
+// newConnectionTimer creates a new timer, after which agent closes its ACS websocket connection
+func newConnectionTimer(client wsclient.ClientServer, connectionTime time.Duration, connectionJitter time.Duration) ttime.Timer {
+	expiresAt := retry.AddJitter(connectionTime, connectionJitter)
+	timer := time.AfterFunc(expiresAt, func() {
+		seelog.Infof("Closing ACS websocket connection after %v minutes", expiresAt.Minutes())
+		// WriteCloseMessage() writes a close message using websocket control messages
+		// Ref: https://pkg.go.dev/github.com/gorilla/websocket#hdr-Control_Messages
+		err := client.WriteCloseMessage()
+		if err != nil {
```

Comment:
What happens now that this connection kind of "escaped" the timer... I wonder if we should rely on ACS to terminate the connection in that case, or attempt to force close it.

Reply:
yes if an error is encountered while writing a close message, we would rely on ACS to terminate the connection. 

I kept this in-line with how we deal with closing the connection when no heartbeats are received and agent tries to close the connection: https://github.com/aws/amazon-ecs-agent/blob/master/agent/acs/handler/acs_handler.go#L487